src/watcher.js

```diff
@@ -42,8 +51,23 @@ export default function onSourceChange(
     proxyFileChanges({artifactsDir, onChange, filePath, shouldWatchFile});
   });
 
-  log.debug(`Watching for file changes in ${sourceDir}`);
-  watcher.watch([], [sourceDir], Date.now());
+  log.debug(`Watching for file changes in ${watchFile || sourceDir}`);
+
+  const watchedDirs = [];
+  const watchedFiles = [];
+
+  if (watchFile) {
+    if (fs.existsSync(watchFile) && !fs.lstatSync(watchFile).isFile()) {
+      throw new UsageError(`"${watchFile}" cannot be passed` +
```

Comment:
"cannot be passed" is vague. Could you rephrase the error message so that it is more actionable? E.g. to:

```
throw new UsageError(`"Invalid --watch-file value: ${watchFile}" is not a file`)
```